sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/services/SpringAiOrchestrationService.java

```diff
@@ -270,4 +274,32 @@ public Translation responseFormat() {
 
     return cl.prompt(prompt).call().entity(Translation.class);
   }
+
+  /**
+   * Get a prompt template by name and input parameters.
+   *
+   * @return the chat response containing the prompt template
+   */
+  @Nullable
+  public ChatResponse promptRegistryToSpringAi() {
+    val repository = new InMemoryChatMemoryRepository();
+    val memory = MessageWindowChatMemory.builder().chatMemoryRepository(repository).build();
+    val advisor = MessageChatMemoryAdvisor.builder(memory).build();
+    val cl = ChatClient.builder(client).defaultAdvisors(advisor).build();
+
+    val promptResponse =
+        new PromptClient()
+            .parsePromptTemplateByNameVersion(
+                "categorization",
+                "0.0.1",
+                "java-e2e-test",
+                "default",
+                false,
+                PromptTemplateSubstitutionRequest.create()
+                    .inputParams(Map.of("inputExample", "I love football")));
+
+    List<Message> messages = SpringUtil.promptRegistryToSpringAi(promptResponse);
```

Comment:
(Minor)

Please improve method name.
* e.g. `promptTemplateToMessages(..)` (you could omit "Spring" in method name, since you are in Spring utility class)

Please find a distinct utility class name.
* e.g. -